sdk/scheduler.go

```diff
@@ -660,17 +660,17 @@ func (scheduler *scheduler) write(writeCtx *WriteContext) {
 	wlog.Debug("[scheduler] starting device write")
 
 	// Get the device.
-	device := scheduler.deviceManager.GetDevice(writeCtx.device)
+	device := scheduler.deviceManager.GetDevice(writeCtx.device.id)
```

Comment:
hmm.  I wonder if this is even needed anymore since the read context now holds a reference to the device, so we shouldn't need to look it up by id anymore.

maybe outside  the  scope of the PR, but food for thought.

Reply:
This is a great point. What is the right way to fix this? Let's get this done now rather than fix it later.

Reply:
I think just replacing this  line with

```go
device := writeCtx.device
```

and similarly, for line 615, 
```diff
- device := scheduler.deviceManager.GetDevice(readCtx.Device.id)
+ device := readCtx.device
```